bpfd/src/multiprog/tc.rs

```diff
@@ -122,8 +122,14 @@ impl TcDispatcher {
             self.if_index, self.revision
         );
         let iface = self.if_name.clone();
-        // Add clsact qdisc to the interface. This is harmless if it has already been added.
-        let _ = tc::qdisc_add_clsact(&iface);
+
+        // let's return an error if the interface is not valid for now.
+        if let Err(e) = tc::qdisc_add_clsact(&iface) {
```

Comment:
Unfortunately, this is isn't going to work this easily because Aya returns the same error whether the `ingress` or `clsact` qdisc has already been added on the interface.  So, I think we're going to need to do something like:

```
qdiscs = Get existing qdiscs for interface (if any)
if qdiscs contains clsact
  we're good (no need to re-add the clsact qdisc)
else if qdiscs contains "ingress"
 return an error saying "ingress qdisc has been added on the interface"
else
  call qdisc_add_clsact() to add clsact qdisc.
```

Also unfortunately, I don't think Aya has an API to get the existing qdiscs, so, we'd need to add one.